include/power/power.h

```diff
@@ -159,31 +159,34 @@ void pm_dump_debug_info(void);
 
 #endif /* CONFIG_PM_DEBUG */
 
-#ifdef CONFIG_PM_STATE_LOCK
 /**
- * @brief Disable particular power state
+ * @brief Set a constraint for a power state
```

Comment:
One point is not clear to me on this API. Not sure if this is the right pace to discuss. Please discard this comment if this isn't.
When setting a constraint on a particular state, should we assume the same constraint should be used on "deeper sleep states", or is the caller expected to call all related states where the constrain apply?

What about substates ?

Reply:
I think we need to wait to discuss that, since this PR is just renaming stuff, but:

* I'm not sure if we can assume a strict linear ordering such that if you block a transition to `STANDBY` you implicitly block transition to `SUSPEND_TO_RAM` and other "lower" states.  Need to see what the prior art does.
* yes, constraints are going to have to take substates into account, either explicitly or through some other back door.